# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1244716624 -7200
#      Thu Jun 11 12:37:04 2009 +0200
# Node ID bf6da2fbfa4e05ea788082f76cd0219941b89f24
# Parent  a76f391a3d02a319be3baf22b31e219fcbe14cf1
examples/polynomial: properly handle magic colon

diff --git a/ChangeLog b/ChangeLog
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,8 +1,13 @@
+2009-06-02  Jaroslav Hajek  <highegg@gmail.com>
+
+	* examples/@polynomial/subsref.m: Handle ':' correctly.
+	* examples/@polynomial/subsasgn.m: Ditto.
+
 2009-06-11  Jaroslav Hajek  <highegg@gmail.com>
 
 	* example/@polynomial/numel.m: New method.
 	* example/@polynomial/Makefile.in: Include it.
 
 2009-05-22 Benjamin Lindner <lindnerb@users.sourceforge.net>
 
 	* mkoctfile.cc.in: mask MSVC specific linker flags for mex output
diff --git a/examples/@polynomial/subsasgn.m b/examples/@polynomial/subsasgn.m
--- a/examples/@polynomial/subsasgn.m
+++ b/examples/@polynomial/subsasgn.m
@@ -4,17 +4,21 @@ function p = subsasgn (p, s, val)
   endif
   switch (s(1).type)
     case "{}"
       ind = s(1).subs;
       if (numel (ind) != 1)
         error ("polynomial: need exactly one index");
       else
         if (length (s) == 1)
-          p.poly(ind{1}+1) = val;
+          if (isnumeric (ind{1}))
+            p.poly(ind{1}+1) = val; 
+          else
+            p.poly(ind{1}) = val;
+          endif
         else
           error ("polynomial: chained subscripts not allowed for {}");
         endif
       endif
     case "."
       fld = s(1).subs;
       if (strcmp (fld, "poly"))
         if (length (s) == 1)
diff --git a/examples/@polynomial/subsref.m b/examples/@polynomial/subsref.m
--- a/examples/@polynomial/subsref.m
+++ b/examples/@polynomial/subsref.m
@@ -10,17 +10,21 @@ function b = subsref (a, s)
       else
         b = polyval (fliplr (a.poly), ind{1});
       endif
     case "{}"
       ind = s(1).subs;
       if (numel (ind) != 1)
         error ("polynomial: need exactly one index");
       else
-        b = a.poly(ind{1}+1);
+        if (isnumeric (ind{1}))
+          b = a.poly(ind{1}+1);
+        else
+          b = a.poly(ind{1});
+        endif
       endif
     case "."
       fld = s.subs;
       if (strcmp (fld, "poly"))
         b = a.poly;
       else
         error ("@polynomial/subsref: invalid property \"%s\"", fld);
       endif
